src/platforms/hosted/serial_unix.c

```diff
@@ -43,6 +47,84 @@ static uint8_t read_buffer[READ_BUFFER_LENGTH];
 static size_t read_buffer_fullness = 0U;
 static size_t read_buffer_offset = 0U;
 
+#ifndef _WIN32
+inline int closesocket(const int socket)
+{
+	return close(socket);
+}
+#endif
+
+/* Socket code taken from https://beej.us/guide/bgnet/ */
+static bool try_opening_network_device(const char *const name)
+{
+	if (!name)
+		return false;
+
+	// Maximum legal length of a hostname
+	char hostname[256];
+
+	// The service name or port number
+	char *service_name;
+
+	struct addrinfo addr_hints = {
+		.ai_family = AF_UNSPEC,
+		.ai_socktype = SOCK_STREAM,
+		.ai_protocol = IPPROTO_TCP,
+		.ai_flags = AI_ADDRCONFIG | AI_V4MAPPED,
+	};
+	struct addrinfo *results;
+
+	// Copy the hostname to an internal array. We need to modify it
+	// to separate the hostname from the service name.
+	if (strlen(name) >= sizeof(hostname)) {
+		DEBUG_WARN("Hostname:port must be shorter than 255 characters\n");
+		return false;
+	}
+	strncpy(hostname, name, sizeof(hostname) - 1U);
+
+	service_name = strstr(hostname, ":");
+	if (service_name == NULL) {
+		DEBUG_WARN("Device name is not a network address in the format hostname:port\n");
+		return false;
+	}
+
+	// Separate the service name / port number from the hostname
+	*service_name = '\0';
+	++service_name;
+
+	if (*service_name == '\0') {
+		return false;
+	}
+
+	if (getaddrinfo(hostname, service_name, &addr_hints, &results) != 0) {
+		return false;
+	}
```

Comment:
These and the other single-statement `if` blocks below can all do with their parens dropping please.